File: src/main/java/Airlines/AirlineTicket/Customer/Models/CustomerService.java
package Airlines.AirlineTicket.Customer.Models;

import Airlines.AirlineTicket.Customer.Customer;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public interface CustomerService {
    Customer saveCustomer(Customer customer);
    List<Customer> getAllCustomers();
    Customer getCustomerById(Long id);
    Customer updateCustomer(Long id, Customer customerDetails);
    void deleteCustomer(Long id);
}

File: frontend/src/services/CustomerService.js
import axios from 'axios';

const CUSTOMER_API_BASE_URL = "http://localhost:8080/customers";

class CustomerService {
    getAllCustomers() {
        return axios.get(CUSTOMER_API_BASE_URL);
    }

    getCustomerById(id) {
        return axios.get(`${CUSTOMER_API_BASE_URL}/${id}`);
    }

    createCustomer(customer) {
        return axios.post(CUSTOMER_API_BASE_URL, customer);
    }

    updateCustomer(id, customer) {
        return axios.put(`${CUSTOMER_API_BASE_URL}/${id}`, customer);
    }

    deleteCustomer(id) {
        return axios.delete(`${CUSTOMER_API_BASE_URL}/${id}`);
    }
}

export default new CustomerService();
